x/meshsecurity/types/keys.go

```diff
@@ -112,10 +114,18 @@ func BuildPipedValsetOpKey(op PipedValsetOperation, val sdk.ValAddress, slashInf
 		copy(r[pn+an+1+1:], b)
 		binary.BigEndian.PutUint64(b, uint64(slashInfo.Power))
 		copy(r[pn+an+1+1+8:], b)
+
 		tn := len(slashInfo.TotalSlashAmount)
 		r[pn+an+1+1+8+8] = byte(tn)
 		copy(r[pn+an+1+1+8+8+1:], slashInfo.TotalSlashAmount)
-		copy(r[pn+an+1+1+8+8+1+tn:], slashInfo.SlashFraction)
+
+		sn := len(slashInfo.SlashFraction)
+		r[pn+an+1+1+8+8+1+tn] = byte(sn)
+		copy(r[pn+an+1+1+8+8+1+tn+1:], slashInfo.SlashFraction)
+
+		timeUnix := slashInfo.TimeInfraction.Unix()
+		binary.BigEndian.PutUint64(b, uint64(timeUnix))
+		copy(r[pn+an+1+1+8+8+1+tn+1+sn:], b)
```

Comment:
Seems clear a better protocol is needed here btw.

Reply:
I agree, but it's available now so I'll still use it

Reply:
No, this has to be implemented. Passing a serialised struct instead of this ugly naked byte array would do, by the way.